dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/NumDependentTestsFigureGenerator.java

```diff
@@ -0,0 +1,414 @@
+        sortList(projList, sortedList, XMLSECURITY_NAME);
+        int[] total;
+        if (((ProjectNumDependentTests) sortedList.get(0)).isFig9()) {
+            total = new int[8];
+        } else if (((ProjectNumDependentTests) sortedList.get(0)).isFig8()) {
+            total = new int[6];
+        } else {
+            total = new int[4];
+        }
+        for (Project temp2 : sortedList) {
+            ProjectNumDependentTests temp = (ProjectNumDependentTests) temp2;
+            // get the correct orig_value...one of the Lists will not have the correct value (will be 0)
+            if (temp.isFig9()) {
+                if (type.equals("orig")) {
+                    latexString +=
+                            generate9(temp.getName(), temp.get_fig9_human_orig(), temp.get_fig9_human_time(), total);
+                } else {
+                    latexString +=
+                            generate9(temp.getName(), temp.get_fig9_auto_orig(), temp.get_fig9_auto_time(), total);
+                }
+                latexString += "\r\n";
+            } else if (temp.isFig8()) {
+                if (type.equals("orig")) {
+                    latexString += generate8(temp.getName(), temp.get_fig8_human(), total);
+                } else {
+                    latexString += generate8(temp.getName(), temp.get_fig8_auto(), total);
+                }
+                latexString += "\r\n";
+            } else if (temp.isFig7()) {
+                if (type.equals("orig")) {
+                    latexString += generate7(temp.getName(), temp.get_fig7_human(), total);
+                } else {
+                    latexString += generate7(temp.getName(), temp.get_fig7_auto(), total);
+                }
+                latexString += "\r\n";
+            }
+        }
+        latexString += "Total";
+        for (int i = 0; i < total.length; i++) {
+            latexString += " & ";
+            latexString += total[i];
+        }
+        latexString += " \\\\";
+
+        // take off the "\r\n" from the last line
+        return latexString;
+    }
+
+    /**
+     * @param args
+     */
+    public static void main(String[] args) {
+        List<String> argsList = new ArrayList<String>(Arrays.asList(args));
+
+        String directoryName = getArgName(argsList, "-directory");
+        // name of directory where files should be outputted
+        String outputDirectoryName = getArgName(argsList, "-outputDirectory");
+
+        /*
+         * 7: prioritization
+         * 8: selection
+         * 9: parallelizaiton
+         */
+
+        File directory = new File(directoryName);
+        File[] fList = directory.listFiles();
+        // create a list of project Objects that each have a diff project name
+        int size = 5;
+        List<Project> proj_orig_arrayList = new ArrayList<Project>(size);
+        List<Project> proj_auto_arrayList = new ArrayList<Project>(size);
+
+        for (File file : fList) {
+            if (file.isFile()) {
```

Comment:
You should be able to refactor this code into a method and have both this class and the EnhancedResultsFigureGenerator call such method.